hs_core/views/__init__.py

```diff
@@ -1719,6 +1707,41 @@ def _set_resource_sharing_status(user, resource, flag_to_set, flag_value):
     return None
 
 
+class FindGroupsView(TemplateView):
+    template_name = 'pages/groups-unauthenticated.html'  # default view is for users not logged in
+
+    def dispatch(self, *args, **kwargs):
+        if self.request.user.is_authenticated():
+            self.template_name = 'pages/groups-authenticated.html'  # update template if user is logged in
+        return super(FindGroupsView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        if self.request.user.is_authenticated():
```

Comment:
isn't this view for unauthenticated users?

Reply:
Have another look at how the context for authenticated/unauthenticated works. There's a default setting and this is changed, if unauthenticated. I think this is all as-intended...